pm/sendermonitor.go

```diff
@@ -280,9 +284,106 @@ func (sm *senderMonitor) cleanup() {
 		if unixNow()-v.lastAccess > int64(sm.ttl) {
 			// Signal the ticket queue consumer to exit gracefully
 			v.done <- struct{}{}
-
+			v.subScope.Close() // close the maxfloat subscriptions
 			delete(sm.senders, k)
 			sm.smgr.Clear(k)
 		}
 	}
 }
+
+func (sm *senderMonitor) redeemWinningTicket(ticket *SignedTicket) (returnErr error) {
+	maxFloat, err := sm.MaxFloat(ticket.Ticket.Sender)
+	if err != nil {
+		returnErr = err
+		return
+	}
+
+	// if max float is zero, there is no claimable reserve left or reserve is 0
+	if maxFloat.Cmp(big.NewInt(0)) <= 0 {
+		if err := sm.QueueTicket(ticket); err != nil {
+			returnErr = err
+			return
+		}
+		returnErr = errors.Errorf("max float is zero")
+		return
+	}
+
+	// If max float is insufficient to cover the ticket face value, queue
+	// the ticket to be retried later
+	if maxFloat.Cmp(ticket.Ticket.FaceValue) < 0 {
+		if err := sm.QueueTicket(ticket); err != nil {
+			returnErr = err
+			return
+		}
+		returnErr = fmt.Errorf("insufficient max float sender=%v faceValue=%v maxFloat=%v", ticket.Ticket.Sender.Hex(), ticket.Ticket.FaceValue, maxFloat)
+		return
+	}
+
+	// Subtract the ticket face value from the sender's current max float
+	// This amount will be considered pending until the ticket redemption
+	// transaction confirms on-chain
+	if subErr := sm.subFloat(ticket.Ticket.Sender, ticket.Ticket.FaceValue); subErr != nil {
+		if err := sm.QueueTicket(ticket); err != nil {
+			returnErr = err
+			return
+		}
+		err = subErr
+		return
+	}
+
+	defer func() {
+		// Add the ticket face value back to the sender's current max float
+		// This amount is no longer considered pending since the ticket
+		// redemption transaction either confirmed on-chain or was not
+		// submitted at all
+		//
+		// TODO(yondonfu): Should ultimately add back only the amount that
+		// was actually successfully redeemed in order to take into account
+		// the case where the ticket was not redeemd for its full face value
+		// because the reserve was insufficient
+		if err := sm.addFloat(ticket.Ticket.Sender, ticket.Ticket.FaceValue); err != nil {
+			returnErr = err
```

Comment:
I actually think returning the error would be preferable especially since `subFloat` also returns an error now. 

Alternatively have the caller of these two methods (`sm.redeemWinningTicket`) be responsible for pushing these max float updates to the channel.

Reply:
> I actually think returning the error would be preferable especially since subFloat also returns an error now.

But, the way the code is currently structured, a tx submission failure error would be overwritten by an `addFloat()` error such that the tx submission failure error would not be logged/exposed to the user. How would you address that?

> Alternatively have the caller of these two methods (sm.redeemWinningTicket) be responsible for pushing these max float updates to the channel.

The `maxFloat()` call could still return an error and that would need to be handled after `addFloat()` is called so I'm not sure if this addresses the problem of overwriting errors in the `defer` statement.

Reply:
fixed in eed4c5e6

I also changed the behaviour of the subscription in 
d2d3fd1 to make the subscriber responsible for calling `LocalSenderMonitor.MaxFloat` to get the latest value when receiving an update notification.

Reply:
Hmm I added a change in the `eth/senderWatcher` by accident in that second commit but I don't think I can undo it without rebasing.

Reply:
> I also changed the behaviour of the subscription in
d2d3fd1 to make the subscriber responsible for calling LocalSenderMonitor.MaxFloat to get the latest value when receiving an update notification.

Why the change?

> Hmm I added a change in the eth/senderWatcher by accident in that second commit but I don't think I can undo it without rebasing.

You mean in eed4c5e? You could push a commit that removes the change.

Reply:
> Why the change?

I felt it made slightly more sense to make the subscriber responsible for getting the max float value and the subscription to just notify when the state changes. 

While in essence there is little difference it does allow us to revert to the original `subFloat` and `addFloat` behaviour without an additional `maxFloat` call and instead offloading that behaviour to the receiver of the channel. 

this would also be true for the watchers `watchSenderInfoChange` and `watchPoolSizeChange` (coming in a subsequent commit) , they can just notify instead of sending updated values.